cli/index.ts

```diff
@@ -802,3 +760,153 @@ export function runCli() {
     yargs.showHelp();
   }
 }
+
+class ProjectConfigOverride {
```

Comment:
this structure seems really odd to me, IMO. i am not sure why we are (partially? why are some options both here and outside?) moving options into this class. I buy the need for a utility function which can construct a final project config "override", but the rest feels quite weird.

Reply:
Namespacing is the goal, to differentiate things like `defaultDatabaseOption` and `ProjectConfigOverride.defaultDatabaseOverrideOption` (the latter has override in the name, though the name of the flag is the same).

The idea is that `ProjectConfigOverride` applies to options where `dataform.json` is expected to already exist.

Reply:
I guess I think it is really weird to define two different flags for the same name. Yes I get that they are used in different contexts, but the _flag_, i.e. its name, is the same entity. If one use-case requires more validation than the other, then put it at the callsite?

Reply:
That makes sense - have de-duped the flags.

Have avoided the misleading `If unset, the value from dataform.json is used.` in `$ dataform init` by overriding the description at init instead.